File: RetrofitConnection/app/src/main/java/com/example/retrofitconnection/config/RetrofitConfig.java
package com.example.retrofitconnection.config;

import com.example.retrofitconnection.service.ProfessorService;

import retrofit2.Retrofit;
import retrofit2.converter.jackson.JacksonConverterFactory;

public class RetrofitConfig {

    private Retrofit retrofit;

    public RetrofitConfig() {
        this.retrofit = new Retrofit.Builder()
                .baseUrl("https://professor-allocation.herokuapp.com/")
                .addConverterFactory(JacksonConverterFactory.create())
                .build();
    }

    public ProfessorService getProfessorService() {
        return retrofit.create(ProfessorService.class);
    }

//    public CoursesService getAll() {
//        return retrofit.create(ProfessorService.class);
//    }
}

File: ProfessorAllocation/app/src/main/java/com/domain/professorallocation/RetrofitConfig.kt
package com.domain.professorallocation

import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory

class RetrofitConfig {
    companion object {
        val retrofit = Retrofit.Builder()
            .baseUrl("https://professor-allocation.herokuapp.com/")
            .addConverterFactory(GsonConverterFactory.create())
            .build()
    }
}